src/line.cpp

```diff
@@ -279,204 +279,84 @@ std::vector<tripoint> continue_line(const std::vector<tripoint> &line, const int
-        return _("north");
-    case NORTHEAST:
-        return _("northeast");
-    case EAST:
-        return _("east");
-    case SOUTHEAST:
-        return _("southeast");
-    case SOUTH:
-        return _("south");
-    case SOUTHWEST:
-        return _("southwest");
-    case WEST:
-        return _("west");
-    case NORTHWEST:
-        return _("northwest");
-    case ABOVENORTH:
-        return _("north and above");
-    case ABOVENORTHEAST:
-        return _("northeast and above");
-    case ABOVEEAST:
-        return _("east and above");
-    case ABOVESOUTHEAST:
-        return _("southeast and above");
-    case ABOVESOUTH:
-        return _("south and above");
-    case ABOVESOUTHWEST:
-        return _("southwest and above");
-    case ABOVEWEST:
-        return _("west and above");
-    case ABOVENORTHWEST:
-        return _("northwest and above");
-    case BELOWNORTH:
-        return _("north and below");
-    case BELOWNORTHEAST:
-        return _("northeast and below");
-    case BELOWEAST:
-        return _("east and below");
-    case BELOWSOUTHEAST:
-        return _("southeast and below");
-    case BELOWSOUTH:
-        return _("south and below");
-    case BELOWSOUTHWEST:
-        return _("southwest and below");
-    case BELOWWEST:
-        return _("west and below");
-    case BELOWNORTHWEST:
-        return _("northwest and below");
-    case CENTER:
-        return _("center");
-    }
-    return "BUG. (line.cpp:direction_name)";
-}
-
-std::string direction_name_short(direction dir)
-{
-    switch (dir) {
-    //~ abbreviated direction names
-    case NORTH:
-        return _("N    ");
-    case NORTHEAST:
-        return _("NE   ");
-    case EAST:
-        return _("E    ");
-    case SOUTHEAST:
-        return _("SE   ");
-    case SOUTH:
-        return _("S    ");
-    case SOUTHWEST:
-        return _("SW   ");
-    case WEST:
-        return _("W    ");
-    case NORTHWEST:
-        return _("NW   ");
-    case ABOVENORTH:
-        return _("UP_N ");
-    case ABOVENORTHEAST:
-        return _("UP_NE");
-    case ABOVEEAST:
-        return _("UP_E ");
-    case ABOVESOUTHEAST:
-        return _("UP_SE");
-    case ABOVESOUTH:
-        return _("UP_S ");
-    case ABOVESOUTHWEST:
-        return _("UP_SW");
-    case ABOVEWEST:
-        return _("UP_W ");
-    case ABOVENORTHWEST:
-        return _("UP_NW");
-    case BELOWNORTH:
-        return _("DN_N ");
-    case BELOWNORTHEAST:
-        return _("DN_NE");
-    case BELOWEAST:
-        return _("DN_E ");
-    case BELOWSOUTHEAST:
-        return _("DN_SE");
-    case BELOWSOUTH:
-        return _("DN_S ");
-    case BELOWSOUTHWEST:
-        return _("DN_SW");
-    case BELOWWEST:
-        return _("DN_W ");
-    case BELOWNORTHWEST:
-        return _("DN_NW");
-    case CENTER:
-        return _("CE");
+    return point(0, 0);
+}
+
+std::string const& direction_name(direction const dir, bool const short_name)
+{
+    enum : int { size = 3*3*3 };
```

Comment:
What is so bad about a switch?

The very good thing about it is: the compiler will warn you if there is ever the case some value of the enum is not handled and it will prevent duplicated cases. This implementation basically erases both advantages.

And I'd assumed the compiler would have generated code very similar to this anyway.